workflow-controller/controller-service/src/main/java/io/hyscale/controller/validator/impl/DockerValidator.java

```diff
@@ -0,0 +1,59 @@
+/**
+ * Copyright 2019 Pramati Prism, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.hyscale.controller.validator.impl;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import io.hyscale.builder.core.models.ImageBuilderActivity;
+import io.hyscale.commons.commands.CommandExecutor;
+import io.hyscale.commons.commands.provider.ImageCommandProvider;
+import io.hyscale.commons.exception.HyscaleException;
+import io.hyscale.commons.logger.WorkflowLogger;
+import io.hyscale.commons.validator.Validator;
+import io.hyscale.controller.model.WorkflowContext;
+import io.hyscale.controller.util.RegistryAndDockerValidatorUtil;
+
+@Component
+public class DockerValidator implements Validator<WorkflowContext> {
+	private static final Logger logger = LoggerFactory.getLogger(DockerValidator.class);
+
+	@Autowired
+	private ImageCommandProvider commandGenerator;
+
+	@Override
+	public boolean validate(WorkflowContext context) throws HyscaleException {
```

Comment:
In case of validation not required, we should return true
This will run for each service spec passed, we should store value in memory and reuse it once docker is validated first time.

Reply:
@SameeraGrandhi Your opinion?

Reply:
This should be executed only once for each command. Not necessary we run throught every service spec. Since this is a post validation, we could just do after processing all the servicespecs before starting deployment workflow.